### ISSUE TITLE
Extending custom motorway/road shields support

### ISSUE THREAD

**Issue by radoslav-orlov** [tracestrack/tracestrack-topo-map#12]
Hi everyone,

While browsing the maps I noticed that countries like Turkey, Greece, etc. have custom icons rendered on top of their motorways/roads/E routes. Would it be possible to extend this to Bulgaria too? In Wikipedia there are SVG icons of [A1-A8](https://commons.wikimedia.org/wiki/Category:Diagrams_of_motorway_number_signs_of_Bulgaria), [all main roads](https://commons.wikimedia.org/wiki/Category:Diagrams_of_national_and_provincial_road_number_signs_of_Bulgaria), and [most class II roads](https://commons.wikimedia.org/wiki/Category:Diagrams_of_national_and_provincial_road_number_signs_of_Bulgaria_II_class), which could be useful. Cheers and thanks in advance.

**Comment by quinncnl** [comment]
Hi! We have plans to open source the road shields svg to support more shields. It'll come soon.

**Comment by quinncnl** [comment]
An email report related to Philippine road shields:
```
* If the reference number is only numbers, it will render as a white shield. Example "170"
* If the reference number starts with an E then a number without space, it will render as a yellow shield. Example "E1"
* If the reference number is AH26 (applies for both ref and int_ref) it applies as a blue rectangle.
```
Ideally we should develop and open source some tools to make adding custom road shields easier. For small scale we can still do manually.

**Comment by computerfan0** [comment]
In Ireland, regional and local roads are currently rendered with default shields. I know it's a minor change (especially for local roads), but the correct shields consist of a white rectangle with black borders and text, [as shown here](https://commons.wikimedia.org/wiki/File:IRL_R101.svg).

Similarly, in the UK, B-roads are rendered with the default shields. A roads show green shields with yellow text. Although primary A-roads (marked as highway=trunk) do have green shields, non-primary A-roads and B-roads (marked as highway=primary and highway=secondary respectively) have white rectangular shields with black borders/text, [as demonstrated here](https://commons.wikimedia.org/wiki/File:UK_road_A1000.svg). Note that highway=primary N-roads in Ireland DO have green shields, so do not change them to have white signs.

**Comment by 1ec5** [comment]
> Ideally we should develop and open source some tools to make adding custom road shields easier.

In case you haven’t seen it, [OSM Americana](https://github.com/ZeLonewolf/openstreetmap-americana/) has developed an extensive collection of shield artwork and a reusable library for drawing them in JavaScript. The countries mentioned above are all covered. It’s designed for vector tiles and requires the tileset to expose details from route relations, as OpenMapTiles does. Nevertheless, you’re welcome to make use of the artwork and tag processing in a different environment – it’s all in the public domain.

**Comment by quinncnl** [comment]
@computerfan0 Indeed. Those two were missed in previous iterations. That should be fixed asap.

@1ec5 Thanks for the note. There are indeed quite some nice shield templates we can adapt (for different width). Will work on a mechanism to render/test shields in May.

**Comment by quinncnl** [comment]
closing as more country-specific shields supports have been deployed in production. The approach is using a Postgres function to calculate the correct shield symbol and use AI agents to research road shield symbol svg.
